floodlight/io/sportradar.py

```diff
@@ -0,0 +1,199 @@
+
+    # extract match id
+    mID = events["sport_event"]["id"]
+
+    # create links from home/away to team id and team name
+    teams = ["home", "away"]
+
+    tID_link = {}
+    home_away_link = {}
+    for competitor in events["statistics"]["totals"]["competitors"]:
+        tID_link.update({competitor["id"]: competitor["qualifier"]})
+        home_away_link.update(
+            {competitor["qualifier"]: (competitor["id"], competitor["name"])}
+        )
+
+    # extract periods
+    periods = set(
+        [event["period_name"] for event in timeline if event["type"] == "period_start"]
+    )
+
+    # create team event dict
+    columns = [
+        "eID",
+        "gameclock",
+        "time_stamp",
+        "minute",
+        "second",
+        "pID",
+        "player_name",
+        "tID",
+        "team_name",
+        "mID",
+        "event_name",
+        "home_score",
+        "away_score",
+        "scorer",
+        "assists",
+        "zone",
+        "shot_type",
+        "outcome",
+        "players",
+    ]
+
+    # seems unnecessary
+    segments = [f"HT{period}" for period in periods]
+
+    team_event_lists = {
+        team: {segment: {col: [] for col in columns} for segment in segments}
+        for team in teams
+    }
+
+    period = None
+    # loop
+    for event in timeline:
+        if not period:
+            # get first period
+            if event["type"] == "period_start":
+                period = event["period_name"]
+                segment = f"HT{period}"
+                segment_start = datetime.datetime.fromisoformat(event["time"])
+            else:
+                # skip events before first period starts
+                continue
+        # get new periods
+        else:
+            if event["type"] == "period_start":
+                period = event["period_name"]
+                segment = f"HT{period}"
+                segment_start = datetime.datetime.fromisoformat(event["time"])
+
+        # extract event, player and team ids and names
+        eID = event["id"]
+
+        # add all teams as competitors if no competitor is specified in event
+        competitor = [event["competitor"]] if "competitor" in event else teams
+
+        tID = home_away_link[competitor[0]][0] if len(competitor) == 1 else None
+        team_name = home_away_link[competitor[0]][1] if len(competitor) == 1 else None
+        pID = event["player"]["id"] if "player" in event else None
+        player_name = event["player"]["name"] if pID else None
+
+        event_name = event["type"]
+
+        # extract time codes and matchclock
+        time_stamp = datetime.datetime.fromisoformat(event["time"])
+        time_delta = time_stamp - segment_start
+        gameclock = time_delta.seconds
+        if "match_clock" in event:
+            match_clock = event["match_clock"]
+            minute, second = [int(x) for x in match_clock.split(":")]
+        else:
+            minute, second = (None, None)
```

Comment:
Thanks for bringing this up! This mismatch is due to the fact that the official "match_clock" that runs up until 60:00 minutes net playing time will diverge from the real-world gross time due to game interruptions. In this version I decided to put the gross time in the game-clock column, as this would link accurately to the position data. The minute, second columns contain the net time played and could be used for other calculations, like net time left. 

I see that this is not immediately clear! To fix this, I could create four different match_clock columns, like (minute_gross, second_gross, minute_net, second_net) or describe this in detail in the Notes. What do you think?

Reply:
Makes sense to me. Using gross time for the game clock to keep in sync with the tracking data is the right choice in my opinion, too. Net time as min:sec makes also sense, and I think we should keep them with the current column names as those are protected columns according to our definitions. Maybe just add columns (minute_gross, second_gross) and I think this definitley needs a short paragraph in the "Notes" section.

Reply:
Alright! Also added the columns "minute_gross" and "second_gross" that translate the gameclock column to minutes and seconds and added a detailed description to the notes.